src/components/Charts/EmissionTreeMap.tsx

```diff
@@ -0,0 +1,141 @@
+import type { FC } from 'react'
+import dynamic from 'next/dynamic'
+import { Spinner } from 'flowbite-react'
+
+import { api } from '../../utils/api'
+import { assetAmountToString } from '../../utils/helpers'
+
+const Chart = dynamic(() => import('react-apexcharts'), { ssr: false })
+
+interface EmissionData {
+  x: string
+  y: number
+  companyId: string
+}
+
+interface FossilFuelSeries {
+  fossilFuelClass: 'y' | 'n'
+  data: EmissionData[]
+  name: string
+}
+
+const EmissionTreeMap: FC = () => {
+  const source = api.company.getEmissionsAndFFClass.useQuery(undefined, {
+    refetchOnWindowFocus: false,
+  })
+
+  if (!source.data)
+    return (
+      <div className="text-center">
+        <Spinner color="info" />
+      </div>
+    )
+
+  const options = {
+    tooltip: {
+      y: {
+        formatter: function (val: number) {
+          return assetAmountToString(val) + ' metric tons CO2'
+        },
+      },
+    },
+    fontFamily: 'Klima',
+    legend: {
+      show: true,
+    },
+    colors: ['#FFA902', '#0F81E8', '#FF6112', '#17292E'],
+    dataLabels: {
+      enabled: true,
+      dropShadow: {
+        enabled: false,
+      },
+    },
+    plotOptions: {
+      treemap: {
+        enableShades: false,
+      },
+    },
+    chart: {
+      events: {
+        dataPointSelection: (
+          event: any,
+          chartContext: any,
+          config: { seriesIndex: number; dataPointIndex: number },
+        ) => {
+          const selectedDataPoint =
+            series[config.seriesIndex].data[config.dataPointIndex]
```

Comment:
This is possibly undefined, maybe use `?` operator as mentioned in another comment
```suggestion
          const selectedDataPoint =
            series[config.seriesIndex]?.data[config.dataPointIndex]
```